app/js/components/queryBuilder/queryBuilder.component.js

```diff
@@ -72,8 +72,10 @@ const QueryBuilderComponent = {
        * @return {Array} - return array of added param objects
        */
       vm.removeParamsFromQuery = selectedParams => {
-        const addedParams = vm[`${vm.currentState()}Set`],
-          comparator = vm.currentState() == "mutations" ? "_id" : "acronym";
+        const addedParams = selectedParams[0].acronym
+            ? vm["diseaseSet"]
+            : vm["mutationsSet"],
+          comparator = selectedParams[0].acronym ? "acronym" : "_id";
```

Comment:
Interesting. So you are "sniffing" the first element of `selectedParams` to see if it has an `acronym` attribute, which tells you that it's a disease, rather than relying on the UI state. This is a clever workaround for the problem at hand, but I wonder if there is a better way to do it. This approach feels like a hack to me.

What happens, for instance, if we decide that mutations should have some sort of `acronym` attribute associated with them? That would trigger a nasty and hard to solve bug.

Is there something more intentional and obvious we can use to identify which `selectedParams` are diseases and which are mutations? In a similar scenario, I have used a `type` parameter for this kind of "switching".